Analysis/Tasks/PWGHF/taskD0D0barCorrelation.cxx

```diff
@@ -670,50 +703,58 @@ struct TaskCCbarCorrelationMCGen {
+      //c-cbar correlation dedicated section
+      //if it's c, search for cbar and evaluate correlations.
+      if (particle1.pdgCode() == 4) {
+
+        registry.fill(HIST("hcountD0triggersMCGen"), 0); //to count trigger c quark (for normalisation)
+
+        for (auto& particle2 : particlesMC) {
+          if (cutEtaCandMax >= 0. && std::abs(particle2.eta()) > cutEtaCandMax) {
+            continue;
+          }
+          if (cutPtCandMin >= 0. && std::abs(particle2.pt()) < cutPtCandMin) {
+            continue;
+          }
+          if (particle2.pdgCode() == -4) {
+            //check whether mothers of quark cbar (from associated loop) are still '-4' particles - in that case the cbar quark comes from its own fragmentation, skip it
+            if (particlesMC.iteratorAt(particle2.mother0()).pdgCode() == -4) {
               continue;
-            if (particle2.pdgCode() == -4) {
-              //check whether mothers of quark cbar are still '-4' particles - in that case the cbar quark comes from its own fragmentation, skip it
-              if (particlesMC.iteratorAt(particle2.mother0()).pdgCode() == -4)
-                continue;
-              double eta1 = particle1.eta(), eta2 = particle2.eta(), pt1 = particle1.pt(), pt2 = particle2.pt(), phi1 = particle1.phi(), phi2 = particle2.phi();
-              registry.fill(HIST("hDeltaEtaPtIntMCGen"), eta2 - eta1);
-              registry.fill(HIST("hDeltaPhiPtIntMCGen"), getDeltaPhi(phi2, phi1));
-              registry.fill(HIST("hCorrel2DPtIntMCGen"), getDeltaPhi(phi2, phi1), eta2 - eta1);
-              registry.fill(HIST("hDeltaEtaVsPtMCGen"), pt1, pt2, eta2 - eta1);
-              registry.fill(HIST("hDeltaPhiVsPtMCGen"), pt1, pt2, getDeltaPhi(phi2, phi1));
-              registry.fill(HIST("hDeltaPtDDbarMCGen"), pt2 - pt1);
-              registry.fill(HIST("hDeltaPtMaxMinMCGen"), std::max(particle2.pt(), particle1.pt()) - std::min(particle2.pt(), particle1.pt()));
-            } // end outer if (check cbar)
-          }   // end inner loop
-        }     //end outer if (check c)
-      }       //end c/cbar if
-    }         //end outer loop
+            }
+            double eta1 = particle1.eta(), eta2 = particle2.eta(), pt1 = particle1.pt(), pt2 = particle2.pt(), phi1 = particle1.phi(), phi2 = particle2.phi();
+            double deltaEta = eta2 - eta1;
+            double deltaPhi = getDeltaPhi(phi2, phi1);
+            registry.fill(HIST("hDeltaEtaPtIntMCGen"), deltaEta);
+            registry.fill(HIST("hDeltaPhiPtIntMCGen"), deltaPhi);
+            registry.fill(HIST("hCorrel2DPtIntMCGen"), deltaPhi, deltaEta);
+            registry.fill(HIST("hDeltaEtaVsPtMCGen"), pt1, pt2, deltaEta);
+            registry.fill(HIST("hDeltaPhiVsPtMCGen"), pt1, pt2, deltaPhi);
+            registry.fill(HIST("hDeltaPtDDbarMCGen"), pt2 - pt1);
+            registry.fill(HIST("hDeltaPtMaxMinMCGen"), std::max(particle2.pt(), particle1.pt()) - std::min(particle2.pt(), particle1.pt()));
```

Comment:
`std::abs`

Reply:
Changed as done for previous occurrences

Reply:
Committed updated version which satisfies the new requests for adaptAnalysisTask (#PR5558).
No modifications are needed for the charge()->sign() update (#PR5611)